### PR TITLE
Improve logged exec output readability

### PR DESCRIPTION
**Pull request by alpar-t**
- Split error and out streams and log them separately
- Log everything in a single call to prevent interference from
  other log messages

This should help triage rest test failures.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/LoggedExec.java [hunk 1]
```diff
@@ -21,16 +21,16 @@
...
-            setErrorOutput(output);
+            setErrorOutput(error);
             setIgnoreExitValue(true);
             doLast((unused) -> {
                     if (getExecResult().getExitValue() != 0) {
                         try {
-                            for (String line : output.toString("UTF-8").split("\\R")) {
-                                getLogger().error(line);
-                            }
+                            getLogger().error(output.toString("UTF-8"));
```
Comment (nik9000) [inline buildSrc/src/main/java/org/elasticsearch/gradle/LoggedExec.java]: Maybe spit something out telling me which one is which?

**Review by nik9000 — COMMENTED** [review]
I don't particularly like separating these because it is sometimes quite useful to look at them together but I *really* don't like them getting jumbled together on one line. This seems like a fine approach for that reason.

**Comment by atorok** [comment]
@nik9000 does this look better ?

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**